### DISCUSSION TITLE
Obsidian frontmatter being duplicated

### DISCUSSION THREAD

**Discussion by tobiasvl** [ObsidianPublisher/obsidian-github-publisher#303] [Q&A — Plugin] [answered]
Hey! I've been using this plugin for a few years, and love it, but I haven't used it to publish for a while. Specifically (this might be relevant) I haven't used it since Obsidian introduced "properties".

I updated one of my old notes and published it. Something strange happened: The resulting file in the GitHub repo ended up with a duplicate frontmatter! You can see the commit here: https://github.com/tobiasvl/tobiasvl.github.io/commit/07454f71433a7ebb20e6d6cc2a74d0122bea3be5

I'm not sure, but maybe this is related to the fact that I have a "text replacer" to turn "jekyll_tags:" into "tags:" - this is meant to change the "jekyll_tags" in the frontmatter (which I don't want to be Obsidian tags) into just "tags", which Jekyll uses to tag pages/posts. That worked fine before Obsidian introduced properties. I have tried to put "jekyll_tags" into the new "Convert properties/dataview fields into tags" field instead, but that doesn't seem to help. I'm not sure what else to try.

<details>
  <summary>My settings</summary>

```
{
  "github": {
    "branch": "main",
    "automaticallyMergePR": true,
    "dryRun": {
      "enable": false,
      "folderName": "github-publisher"
    },
    "tokenPath": "%configDir%/plugins/%pluginID%/env",
    "api": {
      "tiersForApi": "Github Free/Pro/Team (default)",
      "hostname": ""
    },
    "workflow": {
      "commitMessage": "[PUBLISHER] Merge",
      "name": ""
    },
    "verifiedRepo": true
  },
  "upload": {
    "behavior": "obsidian",
    "defaultName": "",
    "rootFolder": "",
    "yamlFolderKey": "",
    "frontmatterTitle": {
      "enable": false,
      "key": "title"
    },
    "replaceTitle": [
      {
        "regex": "",
        "replacement": "",
        "type": "title"
      }
    ],
    "replacePath": [
      {
        "regex": "/blog",
        "replacement": "",
        "type": "path"
      }
    ],
    "autoclean": {
      "enable": false,
      "excluded": []
    },
    "folderNote": {
      "enable": false,
      "rename": "index.md",
      "addTitle": {
        "enable": false,
        "key": "title"
      }
    },
    "metadataExtractorPath": ""
  },
  "conversion": {
    "hardbreak": false,
    "dataview": true,
    "censorText": [
      {
        "entry": "jekyll_tags:",
        "replace": "tags:",
        "after": false,
        "inCodeBlocks": false
      },
      {
        "entry": "jekyll_title:",
        "replace": "title:",
        "after": false
      }
    ],
    "tags": {
      "inline": false,
      "exclude": [],
      "fields": [
        "jekyll_tags"
      ]
    },
    "links": {
      "internal": true,
      "unshared": false,
      "wiki": true,
      "slugify": "disable"
    }
  },
  "embed": {
    "attachments": true,
    "overrideAttachments": [],
    "keySendFile": [],
    "notes": false,
    "folder": "../assets/images",
    "convertEmbedToLinks": "keep",
    "charConvert": "->"
  }
}
```

</details>

**Comment by Mara-Li** [comment, marked as answer]
You need to reload Obsidian / the plugin, as in #302

**Reply by tobiasvl** [reply]
Thanks, but that didn't work :/ It happened again just now: https://github.com/tobiasvl/tobiasvl.github.io/commit/190bde22705e0582ed50bb1bdd8787fede730f29

**Comment by ZeFish** [comment]
I started having that problem and it seems (for me at least) that the bug is coming from post that have \" double quote in them.

**Reply by Mara-Li** [reply]
Well. The bug was oddly specific in fact. 
It was caused by the fact that the regex applied before any frontmatter works.
But the frontmatter works detect the frontmatter in cached obsidian, not the edited text. 
When it doesn't found the old frontmatter, the parser think that their is none, and append the frontmatter (its cut the text around the two first `---`).

The frontmatter text détection is not really the best but i can't found any solution to parse the frontmatter from text without using external dependency.

**Reply by ZeFish** [reply]
Oh... I wish I could help... but the fact is that it dosen't happen in all my notes. Now I can tell you that in the notes where I don't have \" or \# it will work after one or two publish

**Comment by ZeFish** [comment]
I'm just adding something here because I came to something weird... there is a frontmatter that is being completely changed. I posted a screenshot. (The user input is the bottom part)
<img width="369" alt="Screenshot 2024-02-18 at 2 09 14 PM" src="https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/24323/2ddd93b2-d812-4776-a2a7-0b4c078b79ab">

If it can help anyone...
